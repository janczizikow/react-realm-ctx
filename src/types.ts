export interface RenderProp<TProps, TElement = any> {
  (props: TProps): React.ReactElement<TElement>;
}

export interface IRealmQueryArgs {
  filtered?: string;
  variables?: any[];
  sorted?: Realm.SortDescriptor[];
}
