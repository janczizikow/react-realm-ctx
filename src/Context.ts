import React from 'react';
import Realm from 'realm';

export interface IReactRealmContext {
  realm: Realm;
}

const ReactRealmContext = React.createContext<IReactRealmContext>(null as any);

ReactRealmContext.displayName = 'ReactRealm';

export {ReactRealmContext};
