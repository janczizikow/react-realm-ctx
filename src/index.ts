export * from './components/RealmProvider';
export * from './components/RealmConsumer';
export * from './hoc/withRealm';
export * from './hooks/useRealm';
export * from './hooks/useRealmObjectsQuery';
