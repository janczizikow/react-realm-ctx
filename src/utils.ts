export function getDisplayName(
  WrappedComponent: React.ComponentType<any>,
): string {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}
