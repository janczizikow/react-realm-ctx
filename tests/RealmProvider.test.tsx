import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import {RealmProvider, RealmConsumer} from '../src';
import {realm} from './test-utils';

describe('<RealmProvider />', () => {
  beforeEach(() => {
    realm.write(() => {
      realm.deleteAll();
    });
  });

  it('adds realm to the context', () => {
    const tree = renderer
      .create(
        <RealmProvider realm={realm}>
          <RealmConsumer>
            {({realm: realmCtx}) => (
              <Text>Todos: {realmCtx.objects('Todo').length}</Text>
            )}
          </RealmConsumer>
        </RealmProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
