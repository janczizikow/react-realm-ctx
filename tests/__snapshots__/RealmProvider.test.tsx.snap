// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealmProvider /> adds realm to the context 1`] = `
<Text>
  Todos: 
  0
</Text>
`;
